src/lib/viewers/controls/color-picker/ColorPickerToggle.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import ColorPickerPalette from './ColorPickerPalette';
+import { AnnotationMode } from '../annotations/types';
+import './ColorPickerToggle.scss';
+
+export type Props = {
+    isActive?: boolean;
+    annotationMode?: AnnotationMode;
+};
+
+export default function ColorPickerToggle({ annotationMode, isActive = false, ...rest }: Props): JSX.Element | null {
+    const [isColorPickerToggled, setIsColorPickerToggled] = useState(false);
+
+    if (annotationMode !== AnnotationMode.DRAWING) {
+        return null;
+    }
+
+    return (
+        <div className="bp-ColorPickerToggle">
+            {isColorPickerToggled && (
+                <ColorPickerPalette
+                    onColorSelect={(): void => {
+                        setIsColorPickerToggled(false);
+                    }}
+                />
+            )}
+            <button
+                className={classNames('bp-ColorPickerToggle-button', {
+                    'bp-is-active': isActive,
+                })}
+                onClick={(): void => setIsColorPickerToggled(!isColorPickerToggled)}
+                type="button"
+                {...rest}
+            >
+                <svg focusable="false" height="16" viewBox="0 0 16 16" width="16" />
```

Comment:
Is this SVG needed? It's currently empty.

Reply:
This SVG is needed - it's for displaying the swatch shown in the toggle button.

Reply:
Can the background color of the button itself be the "swatch"?

Reply:
I think that complicates things a bit in terms of how the styling would work. For example, if I set the background color of the button to be the swatch, I would need to apply some padding so that the swatch in the middle of the button has the background color - the padding portion would have a different background color. We would also need to find a way to apply some kind of inner border that is on the inner edge of the padding and outer edge of the swatch.

I would prefer keeping the svg element here for the swatch implementation but am open to suggestions for how we would be able to have a clean implementation for setting the background color of the button to be the swatch.

Reply:
Can we use a div for the inner element? What value does the SVG add?

Reply:
Yep, we can use the div instead of the svg, I just need an element there for the swatch.